packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -195,6 +201,66 @@ export default class CallDiagnosticMetrics {
     throw Error('Not implemented');
   }
 
+  /**
+   * Submit Media Quality Event
+   * @param args
+   */
+  submitMQE({
+    name,
+    // additional payload to be merged with default payload
+    payload,
+    options,
+  }: {
+    name: MediaQualityEvent['name'];
+    // additional payload to be merged with default payload
+    payload: RecursivePartial<MediaQualityEvent['payload']> & {
+      intervals: MediaQualityEvent['payload']['intervals'];
+    };
+    options: SubmitMQEOptions;
+  }) {
```

Comment:
this a complex input type, please extract it into a separated type, also it would be nice to see specific types for `MediaQualityEvent['payload']` and `MediaQualityEvent['payload']['intervals']`

Reply:
You already abstracted aways some types for example `MediaQualityEventVideoSetupDelayPayload`, so it would be more consistent if you change these too.

I just noticed you are merging 2 types here with `&`. This is one more reason to extract this type 
```ts
type MQEPayload = MediaQualityEvent['payload']
type SubmitMQEPayload = RecursivePartial<MQEPayload> & Pick<MQEPayload, 'intervals'>;
interface SubmitMQEParameters extends Pick<MediaQualityEvent, 'name'> {
   payload: SubmitMQEPayload;
   options: SubmitMQEOptions;
}
```
you can keep `name: MediaQualityEvent['name'];` I just prefer  `extends Pick<MediaQualityEvent, 'name'>` because you do not need to use the square bracket to access to the prop and you can easily extract mode field later.

Reply:
I've changed it to 
```ts
submitMQE({
    name,
    // additional payload to be merged with default payload
    payload,
    options,
  }: {
    name: MediaQualityEvent['name'];
    // additional payload to be merged with default payload
    payload: SubmitMQEPayload;
    options: SubmitMQEOptions;
  }) {
```